ext/js/language/translator.js

```diff
@@ -438,57 +443,90 @@ export class Translator {
      */
     _getAlgorithmDeinflections(text, options) {
         const {language} = options;
-        const info = this._textPreprocessors.get(language);
+        const info = this._textProcessors.get(language);
         if (typeof info === 'undefined') { throw new Error(`Unsupported language: ${language}`); }
-        const {textPreprocessors, optionSpace: textPreprocessorOptionsSpace} = info;
+        const {textPreprocessors, preprocessorOptionsSpace, textPostprocessors, postprocessorOptionsSpace} = info;
 
-        /** @type {Map<string, import('language').TextPreprocessorOptions<unknown>>} */
-        const variantSpace = new Map();
-        variantSpace.set('textReplacements', this._getTextReplacementsVariants(options));
-        for (const [key, value] of textPreprocessorOptionsSpace) {
-            variantSpace.set(key, value);
-        }
+        const preprocessorVariantSpace = new Map(preprocessorOptionsSpace);
```

Comment:
![image](https://github.com/themoeway/yomitan/assets/24891609/4ab62df8-6b72-404a-b243-1bb87c23d65b)
Each processor has an array of its possible options, (e.g. `[false, true]`, or `['off', 'direct', 'inverse']` or `[[false, false], [true, false], [true, true]]`. Since we need to run all combinations of the processors, this is like each processor's options being a dimension of a matrix, or an axis of a coordinate system/vector space, and we are getting all of the combinations by traversing all of the cells in the matrix / points in the space.